backend/schema/validate_test.go

```diff
@@ -89,59 +90,59 @@ func TestValidate(t *testing.T) {
 		{name: "ValidIngressRequestType",
 			schema: `
 				module one {
-					verb a(HttpRequest<Empty>) HttpResponse<Empty, Empty>
+					export verb a(HttpRequest<Empty>) HttpResponse<Empty, Empty>
 						+ingress http GET /a
 				}
 			`},
 		{name: "InvalidIngressRequestType",
 			schema: `
 				module one {
-					verb a(Empty) Empty
+					export verb a(Empty) Empty
 						+ingress http GET /a
 				}
 			`,
 			errs: []string{
-				"3:13-13: ingress verb a: request type Empty must be builtin.HttpRequest",
-				"3:20-20: ingress verb a: response type Empty must be builtin.HttpRequest",
+				"3:20-20: ingress verb a: request type Empty must be builtin.HttpRequest",
+				"3:27-27: ingress verb a: response type Empty must be builtin.HttpRequest",
 			}},
 		{name: "IngressBodyTypes",
 			schema: `
 				module one {
-					verb bytes(HttpRequest<Bytes>) HttpResponse<Bytes, Bytes>
+					export verb bytes(HttpRequest<Bytes>) HttpResponse<Bytes, Bytes>
 						+ingress http GET /bytes
-					verb string(HttpRequest<String>) HttpResponse<String, String>
+					export verb string(HttpRequest<String>) HttpResponse<String, String>
 						+ingress http GET /string
-					verb data(HttpRequest<Empty>) HttpResponse<Empty, Empty>
+					export verb data(HttpRequest<Empty>) HttpResponse<Empty, Empty>
 						+ingress http GET /data
 
-					// Invalid types.
```

Comment:
If I add this comment back into the test I get an error where it seems to be trying to parse a `data` instead of a verb. I'm still not sure how to solve this issue, but guessing it has something to do with the parser setup.

```sh
unexpected token \"verb\" (expected \"data\" <ident> (\"<\" TypeParameter (\",\" TypeParameter)* \">\")? \"{\" Field* \"}\" Metadata*)",
```

It seems like the parser is assuming `data` before `verb` if I change this:
```go
declUnion            = []Decl{&Data{}, &Verb{}, &Database{}, &Enum{}, &Config{}, &Secret{}}
```
to have `&Verb{}` before `&Data{}` then this issue goes away, but I hit the issue when parsing `Data` instead. So it seems like maybe the parsers aren't specific enough or something else is happening.

Reply:
We probably need to add `participle.UseLookahead(2)` or maybe 3 now, because we have multiple branches in the parser that are ambiguous. That is, `export data` and `export verb` are both ambiguous at the first token, so without lookahead the parser can't know ahead of time which branch to choose.